osgar/drivers/logzeromq.py

```diff
@@ -74,6 +76,25 @@ def run_output(self):
         socket.close()
         context.term()
 
+    def run_reqrep(self):
+        context = zmq.Context()
+        socket = context.socket(zmq.REQ)
+        socket.RCVTIMEO = int(self.timeout * 1000)  # convert to milliseconds
+        socket.connect(self.endpoint)
+        try:
+            while True:
+                dt, __, data = self.bus.listen()
+                socket.send_string(data)
+                while self.bus.is_alive():
+                    try:
+                        self.bus.publish('response', socket.recv())
+                    except zmq.error.Again:
```

Comment:
This part was discussed separately that maybe it should fail and forward the exeption/problem report to the requester module. The timeout is configurable (yes, constant for all requests). Also note, that this is some kind of "version 0" for single user, which waits for the answer. (I am not sure if the answer is dropped after timeout?)